packages/design-system/src/components/tree/horizontal-shift.ts

```diff
@@ -66,7 +69,11 @@ export const useHorizontalShift = <Data extends { id: string }>({
       placement: shiftPlacement(currentDepth),
     } as const;
 
-    const isDragItem = (item: Data | undefined) => item?.id === dragItem.id;
+    const [dragItemId] = dragItemSelector;
+    const isDragItem = (item: Data | undefined) => item?.id === dragItemId;
+
+    const dropTargetPath = getItemPath(root, data.id);
+    dropTargetPath.reverse();
```

Comment:
maybe it was more ergonomic to keep the order after all? :)

Reply:
It's still legacy path. We convert here to target -> root order. Reverse won't be necessary here soon.